examples/cpp/RK2.cpp

```diff
@@ -0,0 +1,50 @@
+#include "mole.h"  // Include the MOLE library
+#include <iostream>
+#include <cmath>    // For sin()
+#include <iomanip>  // For setprecision
+
+// Function prototype for f(t, y)
+double f(double t, double y);
+
+int main() {
+    double h = 0.1;               // Step size
```

Comment:
If `h` is a constant, it should be declared as constexpr. I also recommend adhering to some style guidelines for naming conventions across files. For instance, you could use: https://google.github.io/styleguide/cppguide.html